pkg/querier/frontend/frontend.go

```diff
@@ -164,34 +164,45 @@ func (f *Frontend) handle(w http.ResponseWriter, r *http.Request) {
 	resp, err := f.roundTripper.RoundTrip(r)
 	queryResponseTime := time.Since(startTime)
 
-	if f.cfg.LogQueriesLongerThan > 0 && queryResponseTime > f.cfg.LogQueriesLongerThan {
+	if err != nil {
+		writeError(w, err)
+	} else {
+		hs := w.Header()
+		for h, vs := range resp.Header {
+			hs[h] = vs
+		}
+		w.WriteHeader(resp.StatusCode)
+		io.Copy(w, resp.Body)
+	}
+
+	if f.cfg.LogQueriesLongerThan == 0 && queryResponseTime > f.cfg.LogQueriesLongerThan {
```

Comment:
This condition seems wrong. If `f.cfg.LogQueriesLongerThan` is zero, then all `queryResponseTimes` will be higher than that.

Changelog says `0` is disabled.

Reply:
Sorry about that. That statement should `!= 0`